node/src/node.cpp

```diff
@@ -112,6 +113,44 @@ class control_slot_t:
     }
 };
 
+struct event_middleware_t {
```

Comment:
Looks like a bit overingeniered - here you need only identity, event, logger and auth - common method would be enough in this case.
LGTM anyway.